charmcraft/commands/build.py

```diff
@@ -353,16 +354,16 @@ def pack_charm_in_instance(
             target_arch=get_host_architecture(),
         )
         base_configuration = providers.get_base_configuration(
-            alias=build_base,
+            alias=build_base_alias,
             instance_name=instance_name,
         )
 
         with self.provider.launched_environment(
             project_name=self.metadata.name,
             project_path=self.charmdir,
             base_configuration=base_configuration,
-            build_base=build_base.value,
             instance_name=instance_name,
+            allow_unstable=(build_on.name != "ubuntu"),
```

Comment:
if centos is unstable, we probably want to emit a warning

Reply:
Added